src/topics/topicActions.js

```diff
@@ -34,3 +36,25 @@ export const fetchTopicsForActiveStream = (narrow: Narrow) => async (
   }
   dispatch(fetchTopics(stream.stream_id));
 };
+
+export const deleteMessagesForTopic = (narrow: Narrow, message: Message | Outbox) => async (
```

Comment:
I should have noticed this during one of the preceding passes, but – why does this function take a `Message | Outbox`? All the information it needs is in the `Narrow`.

(For that matter, it would be fairly simple just to pass the stream name and topic as parameters – there's no need to wrap them up in a `Narrow` object which you then have to check, or assume, to be of the right shape.

`Narrow` *should* be a type we'd want here. Unfortunately it's really a server API type that we've pressed into service for insufficiently-closely-related purposes in the mobile app. :disappointed: We've been discussing its replacement for some time.)